src/storage/src/hummock/sstable/multi_builder.rs

```diff
@@ -73,28 +75,42 @@ where
     last_table_id: u32,
     is_target_level_l0_or_lbase: bool,
     split_by_table: bool,
+    split_weight_by_vnode: u32,
+    /// When vnode of the coming key is greater than `largest_vnode_in_current_partition`, we will
+    /// switch SST.
+    largest_vnode_in_current_partition: usize,
 }
 
 impl<F> CapacitySplitTableBuilder<F>
 where
     F: TableBuilderFactory,
 {
     /// Creates a new [`CapacitySplitTableBuilder`] using given configuration generator.
+    #[allow(clippy::too_many_arguments)]
     pub fn new(
         builder_factory: F,
         compactor_metrics: Arc<CompactorMetrics>,
         task_progress: Option<Arc<TaskProgress>>,
         del_agg: Arc<CompactionDeleteRanges>,
         key_range: KeyRange,
         is_target_level_l0_or_lbase: bool,
-        split_by_table: bool,
+        mut split_by_table: bool,
+        mut split_weight_by_vnode: u32,
     ) -> Self {
         let start_key = if key_range.left.is_empty() {
             UserKey::default()
         } else {
             FullKey::decode(&key_range.left).user_key.to_vec()
         };
 
+        if !is_target_level_l0_or_lbase {
+            split_weight_by_vnode = 0;
+        }
+
+        if split_weight_by_vnode > 0 {
```

Comment:
Shall we need to split sst by table_id when enable `split_weight_by_vnode` ? @Little-Wallace

Reply:
> Shall we need to split sst by table_id when enable `split_weight_by_vnode` ? @Little-Wallace

It's just a virtual flag to help us make code logic more clear.